yt_dlp/extractor/telecinco.py

```diff
@@ -89,25 +89,29 @@ def _parse_content(self, content, url):
         services = config['services']
         caronte = self._download_json(services['caronte'], video_id)
         stream = caronte['dls'][0]['stream']
-        headers = self.geo_verification_headers()
-        headers.update({
-            'Content-Type': 'application/json;charset=UTF-8',
+        headers = {
+            'Referer': url,
             'Origin': re.match(r'https?://[^/]+', url).group(0),
-        })
-        cdn = self._download_json(
+            **self.geo_verification_headers(),
+        }
+        cdn = traverse_obj(self._download_json(
             caronte['cerbero'], video_id, data=json.dumps({
                 'bbx': caronte['bbx'],
                 'gbx': self._download_json(services['gbx'], video_id)['gbx'],
-            }).encode(), headers=headers)['tokens']['1']['cdn']
+            }).encode(), headers={
+                'Content-Type': 'application/json',
+                **headers,
+            }), ('tokens', '1', 'cdn'))
         formats = self._extract_m3u8_formats(
-            stream + '?' + cdn, video_id, 'mp4', 'm3u8_native', m3u8_id='hls')
+            stream + '?' + cdn, video_id, 'mp4', 'm3u8_native', m3u8_id='hls', headers=headers)
 
         return {
+            'http_headers': headers,
```

Comment:
we shouldn't return the geo verification headers in the info dict; this defeats the purpose of `--geo-verification-proxy` since the download itself will be using the proxy then too

Reply:
Thanks for the tip. I updated the code accordlingly. And I have to say the geo-proxy works better for me. faster!

Reply:
tiny nitpick: let's move this below
```suggestion
```